.github/workflows/ci.yml

```diff
@@ -35,10 +35,52 @@ jobs:
           opam pin add . -y --no-action
           opam depext -y geneweb
           opam install -y ./*.opam --deps-only --with-test
-          opam pin ocamlformat 0.24.1
       - name: Make ocamlformat > build/distrib
         run: |
           opam exec -- ocaml ./configure.ml --release
-          opam exec -- make fmt distrib
+          opam exec -- make distrib
       - name: Make CI tests
         run: opam exec -- make ci
+
+  build_nnp:
+    runs-on: ${{ matrix.os }}
+    if: false #this job is temporary skipped
```

Comment:
Should we keep it disabled for now?

Reply:
No, of course not. Sorry for not updating this one, this PR is in its “test state”, I was waiting for discussions about what we really want here. I now think we should build with nnp+ancient for Unix (Linux+macOS) to allow users to have the new -cache-in-memory option (OcamlPro work) available in the release (so that should be done with release CI too). The open question left is if we wan’t to test Ocaml < 4.14.x (longer CI, but we don't loose compatibility, this is important because lot of users asks for builds on old systems on shared hosts). That said, we discovered here that mac-latest (14) don't support older Ocalm version, so we need to split the jobs if we wan't to test 4.08 for example it would be on older macOS. I can do a proposal in one commit for this PR (I kept the commit history in order that you can see the builds vs each commits). 
I propose:
* linux-latest + mac-latest on 4.14.x + nnp.
* windows-latest on 4.14.x (no nnp option nor ancient available at the moment in Opam repo for win).
* linux-latest + windows-latest + macos-12 (11 will be depreciated this month) on 4.08.x.

That is 6 builds jobs instead of 3 currently.

Reply:
@Idir if you agree with this proposal, I clean this PR and push a lone commit to test those six builds on current master. We can optionnaly also test Ocaml 5.1.x for Unix that would make 8 builds (veille technologique).

Reply:
Understood. So I'll mainly add the nnp+ancient testing support here. As I see you also restored the testing for Ocaml 5.1.x, I'll update the PR with 8 builds as stated: 4.08x for all | 4.14.x+nnp linux+mac/4.14.x for win | 5.1.x for linux+mac.